### PR TITLE
Leverage docker cache

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/monitoring.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/config.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/scaling.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/semi-sync.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/users.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/0673fb6/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/160/commits/0673fb68f3948ea77092ed5624c189f09eb64300
image: `perconalab/percona-server-mysql-operator:PR-160-0673fb6`

File: build/Dockerfile [hunk 1]
```diff
@@ -11,6 +12,11 @@ ARG GOOS=linux
 ARG GOARCH=amd64
 ARG CGO_ENABLED=0
 
+COPY api ./api
```
Comment (hors) [inline build/Dockerfile:15]: How many MBs we will save?

**Review by inelpandzic — COMMENTED** [review]
Maybe I'm missing something, but I can't see any difference in size with this.

**Comment by egegunes** [comment]
@hors @inelpandzic it was not about the size but time. since we use multi stage builds final image size won't be changed.

**Comment by inelpandzic** [comment]
> @hors @inelpandzic it was not about the size but time. since we use multi stage builds final image size won't be changed.

Yeah, makes sense. This is timed with a `time` command, like `time make`:
Without cache: `make  2.99s user 3.00s system 1% cpu 6:58.85 total`
With cache: `make 2.84s user 2.40s system 1% cpu 5:24.40 total`

**Review by inelpandzic — APPROVED** [review]

**Review by pooknull — APPROVED** [review]

File: build/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 FROM golang:1.19.0 AS go_builder
 WORKDIR /go/src/github.com/percona/percona-server-mysql-operator
 
-COPY . .
```
Comment (hors) [inline build/Dockerfile:4]: @egegunes I do not see any time build differences  with this line and without :(

**Comment by pooknull** [comment]
@hors can we delete `/usr/lib/debug/usr/bin` folder in this PR until [DISTMYSQL-190](https://jira.percona.com/browse/DISTMYSQL-190) is done?

**Comment by hors** [comment]
> @hors can we delete `/usr/lib/debug/usr/bin` folder in this PR until [DISTMYSQL-190](https://jira.percona.com/browse/DISTMYSQL-190) is done?

@pooknull yes, we can do it as a workaround

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/monitoring.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/scaling.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/service-per-pod.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/users.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/87de71b/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/160/commits/87de71b70bf35dfb74e738405c7285832521e5cf
image: `perconalab/percona-server-mysql-operator:PR-160-87de71b`

**Review by pooknull — CHANGES_REQUESTED** [review]
Please delete `/usr/lib/debug/usr/bin` folder after `percona-mysql-shell` install to reduce image size

**Comment by egegunes** [comment]
@pooknull wow thank you

**Review by pooknull — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/5881e12/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-160/5881e12/config.log) |
| We run 2 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/160/commits/5881e12257c0b27a7866d4e43019c19287f4372e
image: `perconalab/percona-server-mysql-operator:PR-160-5881e12`

**Review by hors — APPROVED** [review]

**Review by inelpandzic — APPROVED** [review]

**Pull request merged**